src/cpp/dds/topic/DataWriterImpl.cpp

```diff
@@ -154,22 +154,38 @@ bool DataWriterImpl::write(
     return create_new_change_with_params(ALIVE, data, params);
 }
 
-bool DataWriterImpl::write(
+ReturnCode_t DataWriterImpl::write(
             void* data,
             const rtps::InstanceHandle_t& handle)
 {
+    if (!handle.isDefined())
+    {
+        return ReturnCode_t::RETCODE_BAD_PARAMETER;
+    }
```

Comment:
The standard talks about a `HANDLE_NIL` that we don't support yet. We define a `c_InstanceHandle_Unknown` for other purposes. If we add support for `HANDLE_NIL` this check may be removed.

Reply:
TODO comment added